codeclimate-wrapper

```diff
@@ -18,32 +18,52 @@ EOF
   exit 1
 }
 
+docker_for_mac() {
+  test -S /var/tmp/docker.sock > /dev/null 2>&1
+}
+
+socket_exists() {
+  test -S $socket_path > /dev/null 2>&1
+}
+
+socket_path() {
```

Comment:
@pbrisbin: Thanks for the tips, no worries about the back and forth.

> Leaving the rest of the script completely as-is, except for replacing the hardcoded socket with `$socket_path`.

To be clear: approaching this problem by changing the socket file location _does not work_ due to the differences between how `docker-machine` and Docker for Mac behave. There are _no_ Docker related environment variables set by Docker for Mac and the Docker Machine executables (`boot2docker` and `docker-machine`) may still exist on the user path, so the conditionals which testing for `command -v docker-machine` and `command -v boot2docker` need to be short circuited early.